.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,23 @@
+name: unit-tests
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  unit-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r ./scripts/python/requirements.txt
+      - name: Test with unittest
+        run: |
+          python -m unittest discover -s scripts/python -fv
```

Comment:
Having the test file locally will work for now but if you are thinking of creating it into a package it will need a different [folder structure](https://packaging.python.org/en/latest/tutorials/packaging-projects/#creating-the-package-files) later on